extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,251 @@
+
+		fvec2_t city_pos = country.get_capital()->get_province_definition().get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+
+	return billboard_pos;
+}
+
+PackedByteArray MapItemSingleton::get_crime_icons() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedByteArray());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedByteArray());
+
+	PackedByteArray icons {};
+
+	for(ProvinceInstance const& prov_inst : instance_manager->get_map_instance().get_province_instances()){
+		ProvinceDefinition const* prov = &prov_inst.get_province_definition();
+		if(prov->is_water()) continue; //billboards dont appear over water, skip
+
+		if(prov_inst.get_crime() == nullptr){
+			icons.push_back(0); //no crime on the province
+		}
+		else {
+			icons.push_back(prov_inst.get_crime()->get_icon());
+		}
+		
+	}
+
+	return icons;
+
+}
+
+PackedByteArray MapItemSingleton::get_rgo_icons() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedByteArray());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedByteArray());
+
+	PackedByteArray icons {};
+
+	for(ProvinceInstance const& prov_inst : instance_manager->get_map_instance().get_province_instances()){
+		ProvinceDefinition const* prov = &prov_inst.get_province_definition();
+		if(prov->is_water()) continue; //billboards dont appear over water, skip
+
+		if(prov_inst.get_rgo_good() == nullptr){
+			icons.push_back(0); //no good on the province
+		}
+		else{
+			icons.push_back(prov_inst.get_rgo_good()->get_index()+1);
+		}
+	}
+
+	return icons;
+
+}
+
+/*
+TODO: National focus isn't implemented yet. It could be done at the country instance, or the province instance
+ So this function just returns dummy data.
+ So in the future...
+ - Return the icon of the current national focus of the state
+ if there is a focus on that state, else return 0 to indicate no focus.
+*/
+
+PackedByteArray MapItemSingleton::get_national_focus_icons() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedByteArray());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedByteArray());
+
+	PackedByteArray icons {};
+
+	for(ProvinceInstance const& prov_inst : instance_manager->get_map_instance().get_province_instances()){
+		ProvinceDefinition const* prov = &prov_inst.get_province_definition();
+		if(prov->is_water()) continue; //billboards dont appear over water, skip
+
+		State const* state = prov_inst.get_state();
+		if(prov->get_identifier() == state->get_capital()->get_identifier()){
```

Comment:
- Check if the state is null, I don't think it should ever be but better to be safe than sorry
- `state->get_capital()` returns a `ProvinceInstance const*`, rather than comparing the identifier strings (slow) you can just compare the pointers: `&prov_inst == state->get_capital()` (you need to use `&prov_inst` rather than `prov` as you're comparing `ProvinceInstance`s rather than `ProvinceDefinition`s, you could use `prov` if you compare against `&state->get_capital()->get_province_definition()` but that's just slower for no gain)